scope.c

```diff
@@ -1846,6 +1846,63 @@ Perl_cx_dump(pTHX_ PERL_CONTEXT *cx)
 #endif	/* DEBUGGING */
 }
 
+void
+Perl_create_sv_destructor(pTHX_ SV *subref, SV *args) {
+    PERL_ARGS_ASSERT_CREATE_SV_DESTRUCTOR;
+    SV *variable = newSV_type_mortal(SVt_IV);
+    MAGIC* magic = sv_magicext(variable, subref, PERL_MAGIC_destruct,
+                               &PL_vtbl_destruct, (char *)args, args ? HEf_SVKEY : 0);
+    magic->mg_flags |= MGf_LOCAL;
+}
+
+int
+Perl_magic_freedestruct(pTHX_ SV* sv, MAGIC* mg) {
+    PERL_ARGS_ASSERT_MAGIC_FREEDESTRUCT;
+    dSP;
+    union {
+        SV   *sv;
+        AV   *av;
+        char *pv;
+    } args_any;
+    IV nargs = 0;
+    if (PL_phase == PERL_PHASE_DESTRUCT) {
+        Perl_warn(aTHX_ "Can't call destructor for 0x%p in global destruction\n", sv);
+        return 1;
+    }
+    /* PUSHSTACKi(PERLSI_MAGIC); */
```

Comment:
The `PUSHSTACK` (and a `POPSTACK` at the end) is well-adviced, otherwise any code that doesn't expect the stack to move from under their feet can get confused by an unexpected free handler.

Reply:
done.